modules/core/src/main/scala/org/tesselation/Main.scala

```diff
@@ -34,48 +35,71 @@ object Main extends IOApp {
       logger.info(s"Config loaded") >>
         logger.info(s"App environment: ${cfg.environment}") >>
         parser.parse[IO](args).flatMap { cli =>
-          SecurityProvider.forAsync[IO].use { implicit securityProvider =>
-            loadKeyPair(cfg.keyConfig).flatMap { keyPair =>
-              KryoSerializer.forAsync[IO](coreKryoRegistrar).use { implicit kryoPool =>
-                DoobieDataSource.forAsync[IO](cfg.dbConfig).use { implicit dataSource =>
-                  DoobieTransactor.forAsync[IO].use { implicit doobieTransactor =>
-                    Supervisor[IO].use { _ =>
-                      (for {
-                        res <- AppResources.make[IO](cfg)
-                        nodeId = PeerId.fromPublic(keyPair.getPublic)
-                        p2pClient = P2PClient.make[IO](res.client)
-                        migrations = Migrations.make[IO]
-
-                        _ <- Resource.eval(migrations.migrate)
-
-                        storages <- Resource.eval(Storages.make[IO])
-                        services <- Resource.eval(Services.make[IO](cfg, nodeId, keyPair, storages))
-                        programs <- Resource.eval(Programs.make[IO](storages, services, p2pClient, nodeId))
-
-                        api = HttpApi.make[IO](storages, services, programs, cfg.environment)
-                        _ <- MkHttpServer[IO].newEmber(ServerName("public"), cfg.httpConfig.publicHttp, api.publicApp)
-                        _ <- MkHttpServer[IO].newEmber(ServerName("p2p"), cfg.httpConfig.p2pHttp, api.p2pApp)
-                        _ <- MkHttpServer[IO].newEmber(ServerName("cli"), cfg.httpConfig.cliHttp, api.cliApp)
+          Random.scalaUtilRandom[IO].flatMap { implicit random =>
+            SecurityProvider.forAsync[IO].use { implicit securityProvider =>
+              loadKeyPair(cfg.keyConfig).flatMap { keyPair =>
+                KryoSerializer.forAsync[IO](coreKryoRegistrar).use { implicit kryoPool =>
+                  DoobieDataSource.forAsync[IO](cfg.dbConfig).use { implicit dataSource =>
+                    DoobieTransactor.forAsync[IO].use { implicit doobieTransactor =>
+                      Supervisor[IO].use { _ =>
+                        (for {
+                          res <- AppResources.make[IO](cfg)
+                          nodeId = PeerId.fromPublic(keyPair.getPublic)
+                          _ <- Resource.eval {
+                            logger.info(s"My peer ID $nodeId")
+                          }
+                          p2pClient = P2PClient.make[IO](res.client)
+                          migrations = Migrations.make[IO]
+                          _ <- Resource.eval(migrations.migrate)
+                          queues <- Resource.eval {
```

Comment:
Imo `Queue` should be created together with implementation of Gossip in such manner:

```scala
def make[F[_]]: F[Gossip[F]] = 
  Queue.unbounded[F, ...].map(make(_))
  
def make[F[_]](queue): Gossip[F]
```

Thanks to that you encapsulate the implementation together with Resources/Queues it needs.
It also follows the pattern we have when something needs some effectful values to instantiate. Another pattern is by implicitly passing Resources but it is not the case for Queue.